File: atividade-9/atividade.py
class AFD:
    def __init__(self):
        self.state = 'q0' 

    def transition(self, char):
        if self.state == 'q0':
            if char == '1':
                self.state = 'q1'
            else:
                self.state = 'q0'
        elif self.state == 'q1':
            if char == '0':
                self.state = 'q2'
            else:
                self.state = 'q1'
        elif self.state == 'q2':
            if char == '1':
                self.state = 'q3'  
            else:
                self.state = 'q0'
        elif self.state == 'q3':
            self.state = 'q3'  

    def is_accepted(self, string):
        self.state = 'q0'  
        for char in string:
            self.transition(char)
        return self.state == 'q3'  

if __name__ == '__main__':
    afd = AFD()
    print(afd.is_accepted("101"))      
    print(afd.is_accepted("1101"))    
    print(afd.is_accepted("1001"))     
    print(afd.is_accepted("001"))     
    print(afd.is_accepted("111"))      
    print(afd.is_accepted("010101"))   

File: atividade-4/atividade.go
package main

import (
	"fmt"
)

type AFD struct {
	state string
}

func (a *AFD) transition(char rune) {
	if a.state == "q0" {
		if char == '0' {
			a.state = "q1"
		}
	}
}

func (a *AFD) isAccepted(input string) bool {
	a.state = "q0" 
	for _, char := range input {
		a.transition(char)
	}
	return a.state == "q1" 
}

func main() {
	afd := AFD{}
	fmt.Println(afd.isAccepted("1"))    
	fmt.Println(afd.isAccepted("0"))    
	fmt.Println(afd.isAccepted("111"))  
	fmt.Println(afd.isAccepted("101"))  
	fmt.Println(afd.isAccepted("1100")) 
}
